src/app/_components/TextLink.tsx

```diff
@@ -1,26 +1,48 @@
+import {
+  BookOpen,
+  GitFork,
+  TwitterLogo,
+  Globe,
+  ArrowUpRight,
+} from '@phosphor-icons/react/dist/ssr'
 import clsx from 'clsx'
 
 import { CustomLink } from '@/components/CustomLink'
 
+import { Icon } from './Icon'
+
 type TextLinkProps = {
   href: string
   className?: string
   children: React.ReactNode
+  iconType?: 'caseStudy' | 'repo' | 'twitter' | 'website'
+  isExternalPage?: boolean
```

Comment:
This means we would have to "manually" pass this prop for every `externalPage`

Reply:
Exactly. Is that an issue?

Reply:
It would be nicer if we could think of a solution where that is not necessary. Perhaps we can leverage the logic in `CustomLink` somehow ..

Reply:
Also happy if you just make this work for the Ecosystem Page without thinking of the wider refactor.